src/screens/NewAnalytics/PaymentAnalytics/PaymentsProcessed/PaymentsProcessed.res

```diff
@@ -76,15 +132,32 @@ let make = (
   React.useEffect(() => {
     getPaymentsProcessed()->ignore
     None
-  }, [])
+  }, [granularity])
 
   <div>
     <ModuleHeader title={entity.title} />
     <Card>
-      <GraphHeader title={graphTitle(paymentsProcessed)} viewType setViewType showTabSwitch=true />
+      <PaymentsProcessedHeader
+        title={graphTitle(paymentsProcessed)}
+        viewType
+        setViewType
+        selectedMetric
+        setSelectedMetric
+        granularity
+        setGranularity
+      />
       <div className="mb-5">
         {switch viewType {
-        | Graph => <LineGraph entity={chartEntity} data={paymentsProcessed} className="mr-3" />
+        | Graph =>
+          <LineGraph
+            entity={chartEntity}
+            config={chartEntity.getObjects(
```

Comment:
we will not pass any props to the graph component.Data manipulation should be done in the particular component and final data should be passed to the graph component.